Backend test flake: test_tasks_scheduled_for_immediate_execution_are_handled_correctly

The backend test `core.platform.taskqueue.cloud_tasks_emulator_test.CloudTasksEmulatorUnitTests.test_tasks_scheduled_for_immediate_execution_are_handled_correctly` [failed in develop](https://github.com/oppia/oppia/runs/3883499577?check_suite_focus=true) with the following error:

```text
======================================================================
FAIL: test_tasks_scheduled_for_immediate_execution_are_handled_correctly (core.platform.taskqueue.cloud_tasks_emulator_test.CloudTasksEmulatorUnitTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/oppia/oppia/core/platform/taskqueue/cloud_tasks_emulator_test.py", line 157, in test_tasks_scheduled_for_immediate_execution_are_handled_correctly
    self.url),
AssertionError: Lists differ: ["Tas[24 chars]_name2 with payload {'param1': 'param2', 'para[183 chars]rl."] != ["Tas[24 chars]_name1 with payload {'param1': 'param1', 'para[183 chars]rl."]

First differing element 0:
"Task[23 chars]_name2 with payload {'param1': 'param2', 'para[60 chars]url."
"Task[23 chars]_name1 with payload {'param1': 'param1', 'para[41 chars]url."

+ ["Task Default in queue queue_name1 with payload {'param1': 'param1', "
+  "'param2': 2, 'param3': None} is sent to dummy_url.",
- ["Task Default in queue queue_name2 with payload {'param1': 'param2', "
? ^

+  "Task Default in queue queue_name2 with payload {'param1': 'param2', "
? ^

-  "'param2': {'arg': 'arg1'}, 'param3': [1, 2, 3]} is sent to dummy_url.",
?                                                                         ^

+  "'param2': {'arg': 'arg1'}, 'param3': [1, 2, 3]} is sent to dummy_url."]
?                                                                         ^

-  "Task Default in queue queue_name1 with payload {'param1': 'param1', "
-  "'param2': 2, 'param3': None} is sent to dummy_url."]
```

--- a/core/platform/taskqueue/cloud_tasks_emulator_test.py
+++ b/core/platform/taskqueue/cloud_tasks_emulator_test.py
@@ -107,20 +107,20 @@ class CloudTasksEmulatorUnitTests(test_utils.TestBase):
             queue_name=self.queue_name1)
 
         self.assertEqual(
-            self.output,
-            [
+            set(self.output),
+            {
                 'Task Default in queue %s with payload %s is sent to %s.' % (
                     self.queue_name1,
                     str(self.payload1),
                     self.url)
-            ]
+            }
         )
 
         self.assertEqual(self.unit_test_emulator.get_number_of_tasks(), 1)
         self.unit_test_emulator.process_and_flush_tasks()
         self.assertEqual(
-            self.output,
-            [
+            set(self.output),
+            {
                 'Task Default in queue %s with payload %s is sent to %s.' % (
                     self.queue_name1,
                     str(self.payload1),
@@ -129,7 +129,7 @@ class CloudTasksEmulatorUnitTests(test_utils.TestBase):
                     self.queue_name2,
                     str(self.payload2),
                     self.url),
-            ]
+            }
         )
         self.assertEqual(self.unit_test_emulator.get_number_of_tasks(), 0)
 
@@ -144,8 +144,8 @@ class CloudTasksEmulatorUnitTests(test_utils.TestBase):
         time.sleep(1)
 
         self.assertEqual(
-            self.output,
-            [
+            set(self.output),
+            {
                 'Task Default in queue %s with payload %s is sent to %s.' % (
                     self.queue_name1,
                     str(self.payload1),
@@ -154,5 +154,5 @@ class CloudTasksEmulatorUnitTests(test_utils.TestBase):
                     self.queue_name2,
                     str(self.payload2),
                     self.url),
-            ]
+            }
         )

